tests/IntegrationTest/InstalledPathsOrderTest.php

```diff
@@ -0,0 +1,119 @@
+        ),
+        'minimum-stability' => 'dev',
+        'prefer-stable'     => true,
+    );
+
+    private $composerConfigB = array(
+        'name'              => 'phpcs-composer-installer/sort-order-test',
+        'require-dev'       => array(
+            'phpcs-composer-installer/multistandard' => '*',
+            'phpcs-composer-installer/dummy-src'     => '*',
+            'phpcs-composer-installer/dummy-subdir'  => '*',
+        ),
+        'minimum-stability' => 'dev',
+        'prefer-stable'     => true,
+    );
+
+    /**
+     * Set up test environment before each test.
+     */
+    protected function set_up()
+    {
+        $this->createTestEnvironment();
+    }
+
+    /**
+     * Clean up after each test.
+     */
+    protected function tear_down()
+    {
+        $this->removeTestEnvironment();
+    }
+
+    /**
+     * Test that the paths registered through the plugin are always registered in the same (sort) order.
+     *
+     * @return void
+     */
+    public function testInstalledPathsAreAlwaysRegisteredInSameOrder()
+    {
+        /*
+         * 1. Install using ConfigA in the Composer global directory.
+         */
+        $this->writeComposerJsonFile($this->composerConfigA, static::$tempGlobalPath);
+
+        // Make sure the plugin runs.
+        $this->assertExecute(
+            'composer global install -v --no-ansi',
+            0,    // Expected exit code.
+            Plugin::MESSAGE_RUNNING_INSTALLER, // Expected stdout.
+            null, // No stderr expectation.
+            'Failed to install dependencies.'
+        );
+
+        /*
+         * 2. Install using ConfigB in the Composer local directory.
+         */
+        $this->writeComposerJsonFile($this->composerConfigB, static::$tempLocalPath);
+
+        // Make sure the plugin runs.
+        $this->assertExecute(
+            sprintf('composer install -v --no-ansi --working-dir=%s', escapeshellarg(static::$tempLocalPath)),
+            0,    // Expected exit code.
+            Plugin::MESSAGE_RUNNING_INSTALLER, // Expected stdout.
+            null, // No stderr expectation.
+            'Failed to install dependencies.'
+        );
+
+        /*
+         * 3. Retrieve the installed paths from both and compare to ensure the order is the same.
+         */
+        $globalPaths = $this->executeCliCommand('"vendor/bin/phpcs" --config-show', static::$tempGlobalPath);
+        $this->assertSame(0, $globalPaths['exitcode'], 'Exitcode for "phpcs --config-show" did not match 0 (global)');
+
+        $localPaths = $this->executeCliCommand('"vendor/bin/phpcs" --config-show', static::$tempLocalPath);
+        $this->assertSame(0, $localPaths['exitcode'], 'Exitcode for "phpcs --config-show" did not match 0 (local)');
+
+        // Get the installed paths setting from the config.
+        $this->assertSame(1, preg_match('`installed_paths:\s+([^\n\r]+)\s+`', $globalPaths['stdout'], $matchGlobal));
```

Comment:
Something irks me about the `assertSame` / `preg_match` combination. Not entirely sure what... Could `assertMatchesRegularExpression` be used? :thinking:

Reply:
Good point. I think I may have created this test before we decided to drop PHP 5.3 support and add the PHPUnit Polyfills. Will fix this up.

Reply:
Actually, looking at it now, I _do_ see why I didn't use `assertMatchesRegularExpression()`: we need to remember the `$matches` for handling and use in the _next_ assertion and `assertMatchesRegularExpression()` doesn't have a way to give us access to that.

Reply:
Rebased & force pushed the fixed commit. The additional `$message` parameters in the last three assertions is the only change.